File: Java/String-3/equalIsNot.java

public class string3 {

	// given a string, return true if the number of appearances of "is" anywhere in the string is equal to the number of appearances of "not" anywhere in the string (case sensitive)
	public boolean equalIsNot(String str) {
		
		boolean b = false;
		int isTimes = 0;
		int notTimes = 0;
		
		for (int i = 0; i < str.length() - 1; i++) {
			if (str.charAt(i) == 'i' && str.charAt(i+1) == 's') {
				isTimes++;
			}
		}
		
		for (int i = 0; i < str.length() - 2; i++) {
			if (str.charAt(i) == 'n' && str.charAt(i+1) == 'o' && str.charAt(i+2) == 't') {
				notTimes++;
			}
		}

		if (isTimes == notTimes)
			b = true;
		
		return b; 
		
	}	
	
	public static void main(String[] args) {
		
		string3 o = new string3();
		
		System.out.println(o.equalIsNot("This is not"));
		System.out.println(o.equalIsNot("This is notnot"));
		System.out.println(o.equalIsNot("noisxxnotyynotxisi"));

	}

}

File: Python/String-3/equal_is_not.py

# given a string, return true if the number of appearances of "is" anywhere in the string is equal to the number of appearances of "not" anywhere in the string (case sensitive)
def equal_is_not(str):
    b = False
    is_times = 0
    not_times = 0

    for x in range(len(str)-1):
        if str[x] == 'i' and str[x+1] == 's':
            is_times += 1
    
    for x in range(len(str)-2):
        if str[x] == 'n' and str[x+1] == 'o' and str[x+2] == 't':
            not_times += 1
    
    if is_times == not_times:
        b = True

    return b

print(equal_is_not('This is not'))
print(equal_is_not('This is notnot'))
print(equal_is_not('noisxxnotyynotxisi'))
